src/main/java/net/snowflake/ingest/streaming/internal/IcebergParquetValueParser.java

```diff
@@ -298,4 +359,177 @@ private static int timeUnitToScale(LogicalTypeAnnotation.TimeUnit timeUnit) {
+  /**
+   * Parses an iterable value based on Parquet 3-level list logical type. Please check <a
+   * href="https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#lists">Parquet
+   * Logical Types#Lists</a> for more details. The parsed value is a list of lists, where each inner
+   * list represents a list of elements in the group. For example, an input {@code [1, 2, 3, 4]}
+   * will be parsed as {@code [[1], [2], [3], [4]]}.
+   */
+  private static ParquetBufferValue get3LevelListValue(
+      Object value,
+      GroupType type,
+      Map<String, RowBufferStats> statsMap,
+      ZoneId defaultTimezone,
+      final long insertRowsCurrIndex,
+      String path) {
+    Iterable<?> iterableVal =
+        DataValidationUtil.validateAndParseIcebergList(path, value, insertRowsCurrIndex);
+    List<Object> listVal = new ArrayList<>();
+    final AtomicReference<Float> estimatedParquetSize = new AtomicReference<>(0f);
+    iterableVal.forEach(
+        element -> {
+          ParquetBufferValue parsedValue =
+              parseColumnValueToParquet(
+                  element,
+                  type.getType(0).asGroupType().getType(0),
+                  statsMap,
+                  defaultTimezone,
+                  insertRowsCurrIndex,
+                  concatDotPath(path, THREE_LEVEL_LIST_GROUP_NAME),
+                  true);
+          listVal.add(Collections.singletonList(parsedValue.getValue()));
+          estimatedParquetSize.updateAndGet(sz -> sz + parsedValue.getSize());
+        });
+    return new ParquetBufferValue(listVal, estimatedParquetSize.get());
+  }
+
+  /**
+   * Parses a map value based on Parquet 3-level map logical type. Please check <a
+   * href="https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#maps">Parquet
+   * Logical Types#Maps</a> for more details. The parsed value is a list of lists, where each inner
+   * list represents a key-value pair in the group. For example, an input {@code {"a": 1, "b": 2}}
+   * will be parsed as {@code [["a", 1], ["b", 2]]}.
+   */
+  private static ParquetBufferValue get3LevelMapValue(
+      Object value,
+      GroupType type,
+      Map<String, RowBufferStats> statsMap,
+      ZoneId defaultTimezone,
+      final long insertRowsCurrIndex,
+      String path,
+      boolean isDescendantsOfRepeatingGroup) {
+    Map<?, ?> mapVal =
+        DataValidationUtil.validateAndParseIcebergMap(path, value, insertRowsCurrIndex);
+    List<Object> listVal = new ArrayList<>();
+    final AtomicReference<Float> estimatedParquetSize = new AtomicReference<>(0f);
+    mapVal.forEach(
+        (k, v) -> {
+          ParquetBufferValue parsedKey =
+              parseColumnValueToParquet(
+                  k,
+                  type.getType(0).asGroupType().getType(0),
+                  statsMap,
+                  defaultTimezone,
+                  insertRowsCurrIndex,
+                  concatDotPath(path, THREE_LEVEL_MAP_GROUP_NAME),
+                  true);
```

Comment:
is this hardcoding intentional? add a comment why we're hardcoding true here but passing in isDescendantsOfRepeatingGroup for parsedValue ?

Reply:
both key and value should be hard coded to true. Fixed and modified the test case.